homework/index.js

```diff
@@ -1,162 +1,164 @@
 'use strict';
 
 {
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+    const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
 
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
-    });
-    return elem;
-  }
-
-  function renderError(error) {
-    const root = document.getElementById('root');
-    createAndAppend('h1', root, { text: error.message });
-  }
-
-  function fetchJSON(url, cb) {
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status >= 200 && xhr.status <= 299) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
-  }
-
-  function Contributions(urlText) {
-    const infoContainer = document.getElementById('info_container');
-    const infoRight = createAndAppend('div', infoContainer, {
-      class: 'info_right',
-      text: 'Contributions:',
-    });
-    fetchJSON(urlText, (err, contributors) => {
-      if (err) {
-        renderError(err);
-        return;
-      }
-      contributors.forEach(contributor => {
-        const ContributorEl = createAndAppend('a', infoRight, {
-          class: 'Contributor_el',
-          href: contributor.html_url,
-          target: '_blank',
+    function createAndAppend(name, parent, options = {}) {
+        const elem = document.createElement(name);
+        parent.appendChild(elem);
+        Object.keys(options).forEach(key => {
+            const value = options[key];
+            if (key === 'text') {
+                elem.textContent = value;
+            } else if (key === 'html') {
```

Comment:
Adding this option waters down the concept of this function. It adds ambiguity on how this function should be used: you can now create child elements through `innerHTML` in addition to creating elements one at the time and explicitly making one the parent of the other.

Reply:
yes , I can now  create child elements through innerHTML.  I think that it gives me more options, and bigger concept . So I can decide what to use and when. 
Sure InnerHTML just when it is really necessary and makes the code shorter.

Reply:
You are of course entitled to your own opinion. I just happen to disagree.